easybuild/easyconfigs/g/gmsh/gmsh-4.12.2-foss-2023b.eb

```diff
@@ -0,0 +1,43 @@
+easyblock = 'CMakeMake'
+
+name = 'gmsh'
+version = '4.12.2'
+
+homepage = 'https://gmsh.info/'
+description = "Gmsh is a 3D finite element grid generator with a build-in CAD engine and post-processor."
+
+toolchain = {'name': 'foss', 'version': '2023b'}
+toolchainopts = {'usempi': True}
+
+source_urls = ['https://%(name)s.info/src/']
+sources = ['%(name)s-%(version)s-source.tgz']
+checksums = ['13e09d9ca8102e5c40171d6ee150c668742b98c3a6ca57f837f7b64e1e2af48f']
+
+builddependencies = [
+    ('CMake', '3.27.6'),
+    ('SWIG', '4.1.1'),
+    ('Eigen', '3.4.0'),
+]
+
+dependencies = [
+    ('Python', '3.11.5'),
+    ('PETSc', '3.20.3'),
+    ('SLEPc', '3.20.1'),
+    ('FLTK', '1.3.9'),
+    ('occt', '7.8.0'),
+]
+
+separate_build_dir = True
+
+configopts = "-DENABLE_BUILD_SHARED=ON -DENABLE_WRAP_PYTHON=ON -DENABLE_METIS=1"
+
+sanity_check_paths = {
+    'files': ['bin/%(name)s', 'bin/onelab.py', 'lib/%(name)s.py', 'lib/libgmsh.%s' % SHLIB_EXT],
+    'dirs': [],
+}
+
+sanity_check_commands = ['%(name)s --help']
+
+modextrapaths = {'PYTHONPATH': ['lib']}
+
```

Comment:
This will be done by CMakePythonPackage and take user settings for that into account